docs: modify to detailed links + add details to running-locally (#11405)
argoproj/argo-workflows: docs/running-locally.md

1. [Dev-container CLI](https://github.com/devcontainers/cli). Install the tool and from the argo-workflow folder do `devcontainer up --workspace-folder .` followed by `devcontainer exec --workspace-folder . /bin/bash` to get a shell where you can start building the code. You can use your choice editor outside the docker image to edit code, the changes are mirrored inside the container. Due to a limitation in the CLI only port 8080 (the Web UI) will get exposed for you to access if you run this way. Other services are usable from the shell inside.

After you enter the container, run:
```
make clean
```

If you want to see Argo Workflows UI in http://localhost:8080:
```
make start UI=true

> ````

Reviewer: > Not sure why these just started showing up now 🤔 Seems to be erroring on code backticks.

there's 4 backticks here instead of 3 -- that's what's causing `make docs-spellcheck` to fail. it's just causing syntax errors in general.

you'll then also need to change "popup" to "pop-up" to fix the spelling.

Author: Thanks for let me know!!

Author: I changed this draft pr to open. Please review when you have the time.

> It might take more than 10 minutes. If you're using [Dev Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) extension in Visual Studio Code, After waiting you'll see a popup saying that "Your application running on port 8080 is available".

Reviewer: ```suggestion

It might take more than 10 minutes. If you're using the Dev Containers extension for Visual Studio Code, you'll see a popup once everything is started, saying "Your application running on port 8080 is available".
```

added a newline (usually required by `markdownlint`), improved grammar ("the"), clarified "After waiting" -> "once everything is started" (also typo with capital A).

also IMO, I don't think we need a second link to the extension since it was just mentioned, but nbd either way.

Reviewer: I believe `markdownlint` will also require syntax highlighting on the code blocks (i.e. "```bash") once it runs

> After you enter the container, run:
> ```
> make clean
> ```
> 
> If you want to see Argo Workflows UI in http://localhost:8080:
> ```
> make start UI=true
> ````
> It might take more than 10 minutes. If you're using [Dev Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) extension in Visual Studio Code, After waiting you'll see a popup saying that "Your application running on port 8080 is available".

Reviewer: ```suggestion
Once you have entered the container, continue to [Developing Locally](#developing-locally)
```

Maybe link downward instead so as to not duplicate?

## Developing locally

Add the following to your `/etc/hosts`:

```text
127.0.0.1 dex
127.0.0.1 minio
127.0.0.1 postgres
127.0.0.1 mysql
127.0.0.1 azurite
```

> 

Reviewer: Should probably _move_ this block into "Requirements" (which is currently the non-devcontainer section I guess?), as _only_ the devcontainer [automatically adds hosts](https://github.com/argoproj/argo-workflows/blob/7f22085d16316bdd77bdb1060455140d9d6e1664/.devcontainer/pre-build.sh#L4).

It does feel like there's more standardization and organization that could go into the local set-up -- please feel free to add follow-up PRs! I'll probably make some myself too. Thanks for bringing attention to this!
